File: dataset/Intermediate/Translation/Avatar/v2-Python-code/atcoder_ABC143_D.py
from bisect import bisect_right, bisect_left
(n, *l) = map(int, open(0).read().split())
l.sort()
triangleCount = 0
CheckerFour = 948
CheckerOne = 511
CheckerThree = 442
CheckerTwo = 609
for i in range(n):
    for j in range(i + 1, n):
        a = l[i]
        b = l[j]
        right = bisect_left(l, a + b)
        left = bisect_right(l, max(a - b, b - a))
        tmp = max(0, right - left)
        if CheckerThree & CheckerTwo:
            if CheckerFour & CheckerOne:
                if left <= i < right:
                    tmp = tmp - 1
        if left <= j < right:
            tmp -= 1
        triangleCount += tmp
print(triangleCount // 3)

File: scripts/GAs/Experiment_Results_v2-Python-code/intermediate/SR/Translation/use_test/CodeLlama-34b-Instruct-hf/Avatar/Python/Java/atcoder_ABC143_D.java
```java
import java.util.Arrays;

public class TriangleCount {
    public static void main(String[] args) {
        int[] l = Arrays.stream(args[0].split(" ")).mapToInt(Integer::parseInt).toArray();
        int n = l.length;
        Arrays.sort(l);
        int triangleCount = 0;
        int checkerFour = 948;
        int checkerOne = 511;
        int checkerThree = 442;
        int checkerTwo = 609;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int a = l[i];
                int b = l[j];
                int right = Arrays.binarySearch(l, a + b);
                int left = Arrays.binarySearch(l, Math.max(a - b, b - a));
                int tmp = Math.max(0, right - left);
                if ((checkerThree & checkerTwo) != 0) {
                    if ((checkerFour & checkerOne) != 0) {
                        if (left <= i && i < right) {
                            tmp = tmp - 1;
                        }
                    }
                }
                if (left <= j && j < right) {
                    tmp -= 1;
                }
                triangleCount += tmp;
            }
        }
        System.out.println(triangleCount / 3);
    }
}
```
Note: The above code uses the `Arrays.binarySearch` method to perform a binary search on the sorted array `l`. This method returns the index of the first element that is equal to the search key, or a negative number if the search key is not found. The negative number can be used to determine the insertion point for the search key, which is what we need in this case.
